simplified-books-covers/src/main/java/org/nypl/simplified/books/covers/BookCoverGenerator.java

```diff
@@ -104,7 +104,10 @@ public BookCoverGenerator(
     final SortedMap<String, String> params = new TreeMap<String, String>();
     params.put("title", NullCheck.notNull(title));
     params.put("author", NullCheck.notNull(author));
-    return URIQueryBuilder.encodeQuery(
-      NullCheck.notNull(URI.create("generated-cover://localhost/")), params);
+
+    return LSHTTPURIQueryBuilder.INSTANCE.encodeQuery(
```

Comment:
You should be able to make this more java friendly with the `@JvmStatic` annotation.

https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#static-methods

Reply:
Yep. I didn't bother here given that we have very little Java code left.